web/src/backend/utils.ts

```diff
@@ -47,6 +48,46 @@ export const protectInternalEndpoint = (
   return true;
 };
 
+/**
+ * Validate a request body against a yup schema and returns an error if applicable
+ */
+export const validateRequestSchema = async <T>({
+  schema,
+  req,
+  res,
+}: {
+  schema: yup.Schema<any>;
+  req: NextApiRequest;
+  res: NextApiResponse;
+}): Promise<
+  { isValid: true; parsedParams: T } | { isValid: false; parsedParams?: null }
+> => {
+  let parsedParams: T;
+
+  try {
+    parsedParams = await schema.validate(req.body);
+  } catch (error) {
+    if (error instanceof yup.ValidationError) {
+      errorValidation("invalid", error.message, error.path || null, res, req);
+      return { isValid: false };
+    }
+
+    logger.error("Unhandled yup validation error.", { error, req });
+
+    errorResponse(
```

Comment:
@paolodamico I think we should invoke this error response outside this function, for example in the API endpoint page file. 

This function name says that this function should validate. And in my opinion, this is the only thing it should do. After we validate we decide what to do with the validation result in the place where we executed this validation function, not inside this function. 

It's also not too obvious when you see something like this 

```js
  if (!isValid || !parsedParams) {
    return;
  }
```
Usually, you expect to return a response (or the `errorResponse` in this case) in the API endpoint page code, not the empty return.